src/auth/middleware/basic/index.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const base64 = require('base-64');
+const { userModel } = require('../../models');
+
+
+module.exports = async (req, res, next) => {
+
+  let basicHeaderParts = req.headers.authorization.split(' ');
+  let encodedString = basicHeaderParts.pop();
+  let decodedString = base64.decode(encodedString);
+  let [username, password] = decodedString.split(':');
+
+  try{
+    const user = await userModel.findOne({ where: { username: username } });
+    const valid = await bcrypt.compare(password, user.password);
+    if(valid) {
+      req.user = user;
+      next();
+      res.status(200).json(user);
```

Comment:
This line is duplicated in your router.js; we probably want to be sending the all good response from there instead of here just to maintain consistency in our code writing.